src/arcaflow_plugin_sdk/test_schema.py

```diff
@@ -448,64 +448,77 @@ class TestSubclass(TestParent):
 
 
 class OneOfTest(unittest.TestCase):
-    def test_assignment(self):
-        @dataclasses.dataclass
-        class OneOfData1:
-            a: str
+    @dataclasses.dataclass
+    class OneOfData1:
+        a: str
 
-        @dataclasses.dataclass
-        class OneOfData2:
-            b: int
+    @dataclasses.dataclass
+    class OneOfData2:
+        b: int
 
-        scope = schema.ScopeType(
+    @dataclasses.dataclass
+    class OneOfDataEmbedded1:
+        type: str
+        a: str
+
+    def setUp(self):
+        self.obj_b: schema.ObjectType = schema.ObjectType(
+            OneOfTest.OneOfData2, {"b": PropertyType(schema.IntType())}
+        )
+        self.scope_basic: schema.ScopeType = schema.ScopeType(
             {
                 "a": schema.ObjectType(
-                    OneOfData1, {"a": PropertyType(schema.StringType())}
+                    OneOfTest.OneOfData1,
+                    {"a": PropertyType(schema.StringType())},
                 ),
-                "b": schema.ObjectType(
-                    OneOfData2, {"b": PropertyType(schema.IntType())}
+                "b": self.obj_b,
+            },
+            "a",
+        )
+        self.scope_embedded: schema.ScopeType = schema.ScopeType(
+            {
+                "a": schema.ObjectType(
+                    OneOfTest.OneOfDataEmbedded1,
+                    {
+                        "type": PropertyType(
+                            schema.StringType(),
+                        ),
+                        "a": PropertyType(schema.StringType()),
+                    },
                 ),
+                "b": self.obj_b,
             },
             "a",
         )
+
+    def test_assignment(self):
         s_type = schema.OneOfStringType(
-            {"a": schema.RefType("a", scope), "b": schema.RefType("b", scope)},
-            scope,
-            "_type",
+            {
+                "a": schema.RefType("a", self.scope_basic),
+                "b": schema.RefType("b", self.scope_basic),
+            },
+            scope=self.scope_basic,
+            discriminator_field_name="_type",
         )
         s_type.discriminator_field_name = "foo"
         self.assertEqual("foo", s_type.discriminator_field_name)
-
         schema.OneOfIntType(
-            {1: schema.RefType(1, scope), 2: schema.RefType(2, scope)},
-            scope,
-            "_type",
+            {
+                1: schema.RefType(1, self.scope_basic),
+                2: schema.RefType(2, self.scope_basic),
+            },
+            scope=self.scope_basic,
+            discriminator_field_name="_type",
         )
```

Comment:
Does this invocation accomplish anything?  E.g., I don't see its returned object being captured and I don't see any assertions being made after it returns.

If all we're doing is hoping that no exception is thrown, that's a pretty low bar.  Presumably the invocation should have some sort of effect or behavior which we can (and should) check.

Reply:
Yeah I'm just going to delete that test since it doesn't seem to serve a purpose.

Reply:
I'm not sure that you should remove this test -- this might be the only place in the tests which invokes this c'tor....  It might be better to come up with some reasonable assertion to confirm that the call actually does what it is supposed to.

Reply:
ah yeah it is the only place the `OneOfIntType` is constructed, so I'll call it somewhere in the test suite